include/zephyr/sys/util_macro.h

```diff
@@ -296,7 +296,8 @@ extern "C" {
  * @brief Like <tt>a == b</tt>, but does evaluation and
  * short-circuiting at C preprocessor time.
  *
- * This however only works for integer literal from 0 to 4095.
+ * This however only works for integer literal from 0 to 4096 (literals with U,
+ * e.g. 0U are also included).
```

Comment:
Maybe we add support for `(1U, 1U)` and `(1, 1U)` as well?

Reply:
I actually realize that what i want is to include support where both arguments use `U`.
Let say that there is a set of defines like here https://github.com/zephyrproject-rtos/zephyr/blob/9fbf5d05fd39c24130e3763f5bfeaf54cf422037/include/zephyr/dt-bindings/power/nordic-nrf-gpd.h#L11

```
#define NRF_GPD_SLOW_MAIN 2U
#define NRF_GPD_SLOW_ACTIVE 1U
#define NRF_GPD_FAST_MAIN 3U
#define NRF_GPD_FAST_ACTIVE1 0U
#define NRF_GPD_FAST_ACTIVE0 4U
```

And another define which is set to one of those values. I want to use it as `IS_EQ(MY_GPD, NRF_GPD_FAST_MAIN)`. I've update patch to that.

Reply:
I will update to support mix too. I realized that if for example one value comes from Kconfig then it will no contain `U` so mixed support my also be welcomed.